### PR TITLE
CARP-24 | Add websocket for live changes

### PR DESCRIPTION
**Pull request by priner**

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -9,20 +30,116 @@ async fn get_exchange_rates(db_path: String) -> anyhow::Result<String> {
...
+            res
+        }
+    };
+    Ok(res)
+}
 
+pub async fn route(req: Request<Body>, db_path: String) -> anyhow::Result<Response<String>> {
     let response = match (req.method(), req.uri().path()) {
-        (&Method::GET, "/health") => builder
-            .header("Content-Type", "application/json")
```
Comment (miksax) [inline src/server.rs:17]: Is there reason why change content/type?

File: src/server.rs [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+    header::{self, AsHeaderName},
+    http::HeaderValue,
+    Body, HeaderMap, Method, Request, Response, StatusCode,
+};
 use sea_orm::Database;
+use std::sync::RwLock;
+use tokio::sync::broadcast;
+use tokio_tungstenite::{tungstenite::protocol, WebSocketStream};
+
+pub static WS_BROADCAST_CHANNEL: RwLock<Option<broadcast::Sender<String>>> = RwLock::new(None);
```
Comment (miksax) [inline src/server.rs:13]: Why not store a list of active clients? You can avoid multiple unwraps, etc. Just send it to the list.
just hint

**Review by miksax — APPROVED** [review]
It looks like that is able to work

nitpick: But too many insecure unwraps

File: src/server.rs (hunk 1 above)
Comment (priner) [reply src/server.rs:17]: no, this was by mistake

**Pull request merged**